lib/CleantalkSP/Common/Helpers/IP.php

```diff
@@ -602,4 +602,85 @@ public static function resolve($ip)
         }
         return $ip;
     }
+
+    /**
+     * Convert in_addr byte string to a binary string.
+     * @param $inet
+     * @return string
+     */
+    public static function convertInetTobBits($inet)
+    {
+        $splitted = str_split($inet);
+        $binaryip = '';
+        foreach ($splitted as $char) {
+            $binaryip .= str_pad(decbin(ord($char)), 8, '0', STR_PAD_LEFT);
+        }
+        return $binaryip;
+    }
+
+
+    /**
+     * Convert long integer mask to a decimal CIDR mask.
+     * @param $long_mask
+     * @return int
+     */
+    public static function convertLongIntMaskToDec($long_mask)
+    {
+        $base = ip2long('255.255.255.255');
+        $mask = 32 - log(($long_mask ^ $base) + 1, 2);
+        return (int)$mask;
+    }
+
+    /**
+     * Convert ipv4 or ipv6 address to four long int separate addresses.
+     * @param $ip
+     * @return array
+     */
+    public static function getFourIPv4FromIP($ip)
+    {
+        return self::convertIPv6ToFourIPv4(self::extendIPv6(self::normalizeIPv6($ip)));
+    }
+
+    /**
+     * Return long int network needles to search for in db.
+     * @param $array_of_ipv4_addresses
+     * @return array
+     */
+    public static function getNetworkNeedles($array_of_ipv4_addresses)
+    {
+        $needles = array();
+        foreach ( $array_of_ipv4_addresses as $key => $current_ipv4 ) {
+            $needles[$key] = implode(
+                ',',
+                array_unique(self::calculateMaskForIPs($current_ipv4, 6, 32))
+            );
+        }
+        return $needles;
+    }
+
+    /**
+     * @param string $ip canonical normalized ipv6 record of ip incoming
+     * @param string $network canonical normalized ipv6 record of network
+     * @param int $mask decimal mask of network
+     * @return false|int 1 if addres in the network, 0 therwise, false on errors
+     */
+    public static function isIpv6AddrInIpv6Network($ip, $network, $mask)
+    {
+        $ip_ptoned = inet_pton($ip);
```

Comment:
inet_pton() function can be undefined. because PHP can be compiled without ipv6 option. see here https://www.php.net/manual/en/function.inet-pton.php#104917